### PR TITLE
E2E Swap tests

### PR DESCRIPTION
**Pull request by monokh**
### Description

Adding e2e test of swap functionality. 

### Submission Checklist :pencil:

- [x] BitcoreRPCProvider: Remove duplicate methods already in BitcoinRPCProvider
- [x] RPC providers: fix format of data returned from getUnusedAddress
- [x] BitcoinJsLibSwapProvider: fix redeeming and setting network
- [x] Tests for single chain flow - node and wallets
- [x] Tests for BTC-ETH swap using node and wallets
- [x] Readme explaining how to run tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
   "scripts": {
     "cover": "nyc --reporter=html --reporter=text npm test",
     "lint": "standard \"**/*.js\" \"**/*.html\" \"**/*.md\"",
-    "test": "cross-env NODE_ENV=test mocha $npm_package_options_mocha",
+    "test": "cross-env NODE_ENV=test mocha test/unit $npm_package_options_mocha",
```
Comment (harshjv) [inline package.json:37]: i am directly using `nyc` so with test, we can also see coverage.

https://github.com/liquality/chainabstractionlayer/commit/103fbe94d6cdbf0d61533b04ad8f25ef370aa77c#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R37

File: test/integration/swap/common.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { expect } from 'chai'
+import MetaMaskConnector from 'node-metamask'
+import { Client, providers, crypto } from '../../../'
```
Comment (harshjv) [inline test/integration/swap/common.js]: Use files from `src` (instead of `dist`) so it can be reported back to coveralls.

**Review by harshjv — CHANGES_REQUESTED** [review]
Left some comments. Haven't tried yet.

File: package.json (hunk 1 above)
Comment (monokh) [reply package.json:37]: The purpose of these tests are only to run locally so it's kind of strange to reflect the code coverage in it. It wouldn't be relevant to someone running it.
Perhaps we need an overall `test` script that runs `nyc, npm run test:unit, npm run test:swap` ?

File: test/integration/swap/common.js (hunk 2 above)
Comment (monokh) [reply test/integration/swap/common.js]: changed.

**Review by harshjv — APPROVED** [review]
LGTM

**Pull request merged**